doc/user/layouts/shortcodes/create-source/syntax-details.html

```diff
@@ -21,6 +22,31 @@
 
 ## Details
 
+### Materialized source details
+
+Materializing a source keeps data it receives in an in-memory
+[index](../../overview/api-components/#indexes), the presence of which makes the
+source directly queryable. In contrast, non-materialized sources cannot process
+`SELECT`s directly; to access the data the source receives, you need to create
```

Comment:
Nit: you could also access the data through a non-`SELECT` statement, such as `TAIL`. Perhaps "to query the data ..."?